backend/src/routes/eventRoutes.test.ts

```diff
@@ -0,0 +1,202 @@
+        buskerId: testBuskerIds[0],
+        title: "test event",
+        type: "test type",
+        coordinates: { lat: 0, lng: 0 },
+      },
+    });
+  });
+
+  test("not found if user not found", async function () {
+    const resp = await request(app).get(`/events/0`);
+    expect(resp.statusCode).toEqual(404);
+  });
+});
+
+/************************************** POST /events */
+
+describe("POST /events", function () {
+  test("works for admins", async function () {
+    const resp = await request(app)
+      .post(`/events/create`)
+      .send({
+        buskerId: testBuskerIds[0],
+        title: "test event title",
+        type: "test event type",
+        coordinates: { lat: 1, lng: 1 },
+      })
+      .set("authorization", `Bearer ${adminToken}`);
+    expect(resp.statusCode).toEqual(201);
+    expect(resp.body).toEqual({
+      event: {
+        id: expect.any(Number),
+        buskerId: testBuskerIds[0],
+        title: "test event title",
+        type: "test event type",
+        coordinates: { lat: 1, lng: 1 },
+      },
+    });
+  });
+
+  test("unauth for users", async function () {});
+
+  //NEEDS JSON SCHEMA
+  //   test("bad request with missing data", async function () {
+  //     const resp = await request(app)
+  //       .post(`/events/create`)
+  //       .send({
+  //         title: "test event title",
+  //       })
+  //       .set("authorization", `Bearer ${adminToken}`);
+  //     expect(resp.statusCode).toEqual(400);
+  //   });
+
+  //NEEDS JSON SCHEMA
+  //   test("bad request with invalid data", async function () {
+  //     const resp = await request(app)
+  //       .post(`/events/create`)
+  //       .send({
+  //         buskerId: testBuskerIds[0],
+  //         title: "test event title",
+  //         type: "test event type",
+  //         coordinates: "not coordinates",
+  //       })
+  //       .set("authorization", `Bearer ${adminToken}`);
+  //     expect(resp.statusCode).toEqual(400);
+  //   });
+});
+
+/************************************** PATCH /events/:id */
+
+describe("PATCH /events/:id", () => {
+  test("works for admins", async function () {
+    const resp = await request(app)
+      .patch(`/events/${testEventIds[0]}`)
+      .send({
+        title: "New title",
+      })
+      .set("authorization", `Bearer ${adminToken}`);
+    expect(resp.body).toEqual({
+      event: {
+        id: testEventIds[0],
+        buskerId: testBuskerIds[0],
+        title: "New title",
+        type: "test type",
+        coordinates: { lat: 0, lng: 0 },
+      },
+    });
+  });
+
+  //CURRENTLY AUTHORIZING EVERYONE TO UPDATE EVENTS. THIS TEST IS NEEDED
+  //WHEN WE INTRODUCE BUSKERS MODEL
+//   test("unauth for others", async function () {
+//     const resp = await request(app)
+//       .patch(`/events/${testEventIds[0]}`)
+//       .send({
+//         title: "New title",
+//       })
+//       .set("authorization", `Bearer ${u1Token}`);
+//     expect(resp.statusCode).toEqual(401);
+//   });
```

Comment:
Similar idea to leave the failing tests for `jsonschema`, might be better to allow this test to fail to remind us to implement this feature